core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -38,7 +38,13 @@ object ConsoleConsumer extends Logging {
 
   def main(args: Array[String]) {
     val conf = new ConsumerConfig(args)
-    run(conf)
+    try {
+      run(conf)
+    } catch {
+      case e: Throwable =>
+        error("Error running consumer: ", e)
+    }
+    System.exit(0);
```

Comment:
@ijuma Yes, I think we are leaking non-daemon threads here. I find that, if consumer is compiled with Java 7 and run with Java 8, when it times out, the above exception is wrong and consumer hang. To fix the problem, I need to both catch the exception and do System.exit(..).

I followed the practice used in ProducerPerformance.scala, which invokes `System.exit(0)` at the end of `main()`. Do you like me to add an additional System.exit(-1) right after `error(...)` in `catch{...}`, and remove `the System.exit(0)`?